src/compute/src/command_channel.rs

```diff
@@ -111,21 +111,11 @@ pub fn render<A: Allocate>(timely_worker: &mut TimelyWorker<A>) -> (Sender, Rece
                         return;
                     };
 
-                    loop {
-                        let (cmd, epoch) = match input_rx.try_recv() {
-                            Ok(msg) => msg,
-                            Err(TryRecvError::Empty) => break,
-                            Err(TryRecvError::Disconnected) => {
-                                // Drop our capability to shut down.
-                                capability = None;
-                                break;
-                            }
-                        };
-
-                        assert_eq!(worker_id, 0);
+                    assert_eq!(worker_id, 0);
 
+                    let input: Vec<_> = input_rx.try_iter().collect();
+                    for (cmd, epoch) in input {
                         let worker_cmds = split_command(cmd, peers)
-                            .into_iter()
                             .map(|cmd| (cmd, epoch))
                             .enumerate();
```

Comment:
You could consider moving the `enumerate` into `split_command`. It relies on the property that `split_command` produces an iterator with exactly `peers` elements, which isn't entirely obvious reading this code.

Reply:
Good idea!